tests/conftest.py

```diff
@@ -159,11 +159,6 @@ def job_manager(app):
     return app.job_manager
 
 
-@pytest.fixture
-def serial():
-    return itertools.count(1)
-
-
```

Comment:
Why did you remove this? The `job_factory` fixture still uses it…

Reply:
I wonder if it wouldn't make more sense to make an undefered job, and then let the manager defer it. The InMemory backend already maintains a counter.

(Anyway, WIP :D )

Reply:
I see. But with this the object provided by the new `job_factory` fixture (that you'd named `deferred_job_factory`) would a coroutine function. So that would imply doing `job = await job_factory(…)` instead of `job = job_factory(…)` everywhere we use that fixture. That's a lot of changes. I'd suggest that, if we want to do that, we do it with a separate PR.